Helper/Image.php

```diff
@@ -61,6 +62,44 @@ public function getUrl()
         return $url;
     }
 
+    protected function initBaseFile()
+    {
+        $model = $this->_getModel();
+        $baseFile = $model->getBaseFile();
+        if (!$baseFile) {
+            if ($this->getImageFile()) {
+                $model->setBaseFile($this->getImageFile());
+            } else {
+                $model->setBaseFile($this->getProductImageType());
+            }
+        }
+        return $this;
+    }
+
+    /**
+     * Configurable::setImageFromChildProduct() only pulls 'image' type
+     * and not the type set by the imageHelper
+     *
+     * @return string
+     */
+    private function getProductImageType()
+    {
+        if (!$this->getImageFile() && $this->getType() !== 'image') {
```

Comment:
To be 100% sure, is it possible to call `$this->getProduct()->getImage();` even if  `!$this->getImageFile() && $this->getType() !== 'image'` ? I've reformulated the following code to highlight what I mean (hopefully).

```suggestion:-0+13
if ($this->getImageFile() || $this->getType() === 'image'
    || $this->getProduct()->getTypeId() !== ProductTypeConfigurable::TYPE_CODE) {
    return $this->getProduct()->getImage();
}

$childProducts = $this->getProduct()->getTypeInstance()->getUsedProducts($this->getProduct());
foreach ($childProducts as $childProduct) {
    if ($childProduct->getData($this->getType())
        && $childProduct->getData($this->getType()) !== 'no_selection') {
        return $childProduct->getData($this->getType());
    }
}

// should it be possible still end up here?
return $this->getProduct()->getImage();
```

Reply:
I do believe that exiting/returning as soon as possible is a good practice, even if it means having several returns. Yet, don't take my word for it, ask around for other opinions.

My very first goal with the comment was to highlight that you could end up with calling $this->getProduct()->getImage(); if there's no image and only "no_selection" type in child products. I wanted to ensure it was a possible / safe possibility.

Reply:
the method `$this->getProduct()->getImage()` will always return a value so it's safe to return as the default. I've updated the code to break it apart. Hopefully it's more readable.

Reply:
It is. LGTM :+1: